oci/client/push.go

```diff
@@ -84,3 +146,28 @@ func (c *Client) Push(ctx context.Context, url, sourceDir string, meta Metadata,
 
 	return ref.Context().Digest(digest.String()).String(), err
 }
+
+// createTarballLayer creates a layer from a gzipped tarball of a directory.
+func (c *Client) createTarballLayer(path string, mediaType types.MediaType, dir string, ignorePaths []string) (gcrv1.Layer, error) {
+	layerFile := filepath.Join(dir, "artifact.tgz")
+	if err := c.Build(layerFile, path, ignorePaths); err != nil {
+		return nil, err
+	}
+	return tarball.LayerFromFile(layerFile, tarball.WithMediaType(mediaType))
```

Comment:
```suggestion
	return tarball.LayerFromFile(layerFile, tarball.WithMediaType(mediaType), tarball.WithCompression(compression.GZip), tarball.WithCompressedCaching)
```

We don't push a plain tarball, but a compressed one.

Reply:
> tarball.WithCompression(compression.GZip)

This is the default compression. I think we can leave it out